packages/nuxt/src/pages/runtime/router.ts

```diff
@@ -182,8 +182,10 @@ export default defineNuxtPlugin(async (nuxtApp) => {
 
   nuxtApp.hooks.hookOnce('app:created', async () => {
     try {
+      const initialRoute = router.resolve(initialURL)
+      delete initialRoute.name
       await router.replace({
-        ...router.resolve(initialURL),
+        ...initialRoute,
         force: true
```

Comment:
Would this work if we just did something like this?

```suggestion
        ...router.resolve(initialURL),
        name: undefined,
        force: true
```

Reply:
yes, vue-router use `if ('name' in location && location.name)` so it works too